File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportJava/Networking/src/main/java/transport/networking/protocol/Response.java
package transport.networking.protocol;

import java.io.Serializable;

public class Response implements Serializable {
    private ResponseType type;
    private Object data;

    public Response(ResponseType type, Object data) {
        this.type = type;
        this.data = data;
    }

    public ResponseType getType() {
        return type;
    }

    public Object getData() {
        return data;
    }
}

File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportC#/Network/protocol/Response.cs
using FirmaTransportC_.Networking.dto;

namespace FirmaTransportC_.Networking.protocol
{
    public class Response
    {
        public ResponseType Type { get; set; }
        public string? ErrorMessage { get; set; }

        public UserDTO? User { get; set; }
        public List<TripDTO>? Trips { get; set; }
        public List<SeatDTO>? Seats { get; set; }

        public override string ToString() => $"Response{{Type={Type}, Error='{ErrorMessage}'}}";
    }
}
